### ISSUE TITLE
*** ERROR: DeriveKey() entered Limbo. when decrypt an Excel 2007 hash

### ISSUE THREAD

**Issue by junhua007** [openwall/john#4780]
### Checklist ###
- [x] :1st_place_medal: I've read and understood these instrucions;
  - This is not a support forum, it's a bug tracker. For questions and support, review postings on the [john-users mailing list](https://www.openwall.com/lists/john-users/).
- [x] :thumbsup: I've tested using latest bleeding version from this repository.
  - Be clear about your environment and what you are doing. Share a sample hash or file that can be used to reproduce.
- [ ] :confused: I'm confused and I need guidance.
  - Please, read the instructions at https://www.openwall.com/john/#lists, then join the list before posting.

### IMPORTANT ###
We expect only reports of issues with the latest revision found in this GitHub repository. We do not expect in here, and have little use for, reports of issues only seen in a release or in a distro package.

Attach details about your OS and about john, including:
- The output of `./john --list=build-info`.
Version: 1.9.0-jumbo-1+bleeding-491483c 2021-08-04 19:17:37 -0300
Build: linux-gnu 64-bit x86_64 AVX AC OMP
SIMD: AVX, interleaving: MD4:3 MD5:3 SHA1:1 SHA256:1 SHA512:1
CPU tests: AVX
$JOHN is ./
Format interface version: 14
Max. number of reported tunable costs: 4
Rec file version: REC4
Charset file version: CHR3
CHARSET_MIN: 1 (0x01)
CHARSET_MAX: 255 (0xff)
CHARSET_LENGTH: 24
SALT_HASH_SIZE: 1048576
SINGLE_IDX_MAX: 32768
SINGLE_BUF_MAX: 4294967295
Effective limit: Max. KPC 32768
Max. Markov mode level: 400
Max. Markov mode password length: 30
gcc version: 4.8.5
GNU libc version: 2.17 (loaded: 2.17)
Crypto library: OpenSSL
OpenSSL library version: 0100020bf
OpenSSL 1.0.2k  26 Jan 2017 (loaded: OpenSSL 1.0.2k-fips  26 Jan 2017)
GMP library version: 6.0.0
File locking: fcntl()
fseek(): fseek
ftell(): ftell
fopen(): fopen
memmem(): System's
times(2) sysconf(_SC_CLK_TCK) is 100
Using times(2) for timers, resolution 10 ms
HR timer: clock_gettime(), latency 40 ns
Total physical host memory: 7821 MiB
Available physical host memory: 4977 MiB
Terminal locale string: en_US.UTF-8
Parsed terminal locale: UTF-8

- The command line you are using.
./john --wordlist=/tmp/passwd.txt  /tmp/hash1.txt

When I run the command to crack the password of an encrypted Excel 2007 file with a word list, john reports the following error and exists:

Using default input encoding: UTF-8
Loaded 1 password hash (Office, 2007/2010/2013 [SHA1 128/128 AVX 4x / SHA512 128/128 AVX 2x AES])
Cost 1 (MS Office version) is 2007 for all loaded hashes
Cost 2 (iteration count) is 50000 for all loaded hashes
Will run 2 OpenMP threads
Press 'q' or Ctrl-C to abort, almost any other key for status
Warning: Only 1 candidate left, minimum 8 needed for performance.

*** ERROR: DeriveKey() entered Limbo.
Please report to john-dev mailing list.

[hash1.txt](https://github.com/openwall/john/files/6946701/hash1.txt)
[passwd.txt](https://github.com/openwall/john/files/6946703/passwd.txt)

**Comment by ghost** [comment]
Thanks for reporting. Your document is not currently supported.

Not sure if magnum needs an example to fix this. We will reply to you sooner or later.

git blames:
```
Now: fprintf(stderr, "\n\n*** ERROR: DeriveKey() entered Limbo.\n");
9 years ago, magnum: fprintf(stderr, "**WARNING: DeriveKey() entered unimplemented branch. False negatives?\n")
10 years ago, Dhiru: /* TODO: finish up this function */
```

**Comment by magnumripper** [comment]
I will likely need a sample excel file that triggers the problem in order to have a chance fixing this. If you could send the file to me in private email I'll treat it with discretion.

**Comment by magnumripper** [comment]
> I will likely need a sample excel file that triggers the problem

I take that back (for a while). I think I have a rough idea what is missing, will try tomorrow with the supplied samples. Again, thank you for reporting!

**Comment by magnumripper** [comment]
Fixed in #4782
